src/uniprotkb/components/entry/CommunityCuration.tsx

```diff
@@ -4,21 +4,171 @@ import {
+  switch (section) {
+    case EntrySection.Function:
+      return (communityAnnotation?: CommunityAnnotation) =>
+        communityAnnotation?.function;
+    case EntrySection.DiseaseVariants:
+    case EntrySection.PhenotypesVariants:
+      return (communityAnnotation?: CommunityAnnotation) =>
+        communityAnnotation?.disease;
+    case EntrySection.NamesAndTaxonomy:
+      return (communityAnnotation?: CommunityAnnotation) =>
+        communityAnnotation?.proteinOrGene;
+    default:
+      return null;
+  }
+};
+
+const groupByCommunityAnnotation = (
+  section: EntrySection,
+  communityReferences: Reference[]
+) => {
+  const getAnnotation = annotationGetter(section);
+  if (!getAnnotation) {
+    return null;
+  }
+  const annotationToCommunityReferences = new Map<string, Reference[]>();
+  for (const communityReference of communityReferences) {
+    const annotation = getAnnotation(communityReference.communityAnnotation);
+    if (annotation) {
+      const r = annotationToCommunityReferences.get(annotation) || [];
+      r.push(communityReference);
+      annotationToCommunityReferences.set(annotation, r);
+    }
+  }
+  return annotationToCommunityReferences;
+};
+
+const SubmissionDate = ({
+  accession,
+  citationId,
+  submissionDate,
+}: {
+  accession: string;
+  citationId?: Citation['id'];
+  submissionDate?: string;
+}) => (
+  <ExternalLink
+    url={
+      citationId && citationId.match(/\d+/)
+        ? externalUrls.CommunityCurationGetByAccessionAndPmid(
+            accession,
+            citationId
+          )
+        : externalUrls.CommunityCurationGetByAccession(accession)
+    }
+  >
+    <time dateTime={submissionDate}>{submissionDate}</time>
+  </ExternalLink>
+);
+
+const GroupedCommunityReference = ({
+  annotation,
+  accession,
+  references,
+  section,
+  databaseInfoMaps,
+}: {
+  annotation: string;
+  accession: string;
+  references: Reference[];
+  section: EntrySection;
+  databaseInfoMaps: DatabaseInfoMaps | null;
+}) => (
+  <Card className={styles['reference-card']}>
+    <h4>
+      {section === EntrySection.NamesAndTaxonomy
+        ? 'Community suggested name: '
+        : ''}
+      {annotation}
+    </h4>
+    <table>
+      <thead>
+        <tr>
+          <th>Source</th>
+          <th>Submission date</th>
+          <th>Contributor</th>
+        </tr>
+      </thead>
+      <tbody>
+        {references
+          .sort((a, b) => {
+            if (
+              !a.communityAnnotation?.submissionDate &&
+              !b.communityAnnotation?.submissionDate
+            ) {
+              return 0;
+            }
+            if (
+              !a.communityAnnotation?.submissionDate ||
+              !b.communityAnnotation?.submissionDate
+            ) {
+              return 1;
+            }
+            return a.communityAnnotation.submissionDate.localeCompare(
+              b.communityAnnotation.submissionDate
+            );
+          })
```

Comment:
Are we sure this works as expected? (See first screenshot in this other comment https://github.com/ebi-uniprot/uniprot-website/pull/828/files#r1844271314).

It might be good to extract the function to:
- Give it a name (and potentially comment the function) that conveys the intent
- Test it
- (Marginally) improve performance by defining it only once instead of at every render

Also, this is a mutation of the props passed to the component, so a bit of a red flag in React world. Instead I'd usually do `Array.from(list).sort()` to first copy then mutate-sort the array, but we could also start using the fairly recently added (ES2023) non-mutating methods and in this case [use `toSorted` instead](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSorted).

Reply:
Good call to extract into a separate function as wasn't working as expected when looking at the tests.